src/det.jl

```diff
@@ -1,49 +1,26 @@
-@inline det(A::StaticMatrix) = _det(Size(A), A)
-@inline logdet(A::StaticMatrix) = _logdet(Size(A), A)
+@inline function det(A::StaticMatrix)
+    T = eltype(A)
+    S = typeof((one(T)*zero(T) + zero(T))/one(T))
+    _det(Size(A),A,S)
+end
 
-@inline _det(::Size{(1,1)}, A::StaticMatrix) = @inbounds return A[1]
+@inline logdet(A::StaticMatrix) = log(det(A))
 
-@inline function _det(::Size{(2,2)}, A::StaticMatrix)
-    @inbounds return A[1]*A[4] - A[3]*A[2]
-end
+@inline _det(::Size{(1,1)}, A::StaticMatrix,::Type) = @inbounds return A[1]
 
-@inline function _det(::Size{(2,2)}, A::StaticMatrix{<:Any, <:Any, <:Unsigned})
-    @inbounds return Signed(A[1]*A[4]) - Signed(A[3]*A[2])
+@inline function _det(::Size{(2,2)}, A::StaticMatrix, S::Type)
+    A = similar_type(A,S)(A)
+    @inbounds return A[1]*A[4] - A[3]*A[2]
 end
 
-@inline function _det(::Size{(3,3)}, A::StaticMatrix)
+@inline function _det(::Size{(3,3)}, A::StaticMatrix, S::Type)
+    A = similar_type(A,S)(A)
     @inbounds x0 = SVector(A[1], A[2], A[3])
     @inbounds x1 = SVector(A[4], A[5], A[6])
     @inbounds x2 = SVector(A[7], A[8], A[9])
     return vecdot(x0, cross(x1, x2))
 end
 
-@inline function _det(::Size{(3,3)}, A::StaticMatrix{<:Any, <:Any, <:Unsigned})
-    @inbounds x0 = SVector(Signed(A[1]), Signed(A[2]), Signed(A[3]))
-    @inbounds x1 = SVector(Signed(A[4]), Signed(A[5]), Signed(A[6]))
-    @inbounds x2 = SVector(Signed(A[7]), Signed(A[8]), Signed(A[9]))
-    return vecdot(x0, cross(x1, x2))
+@inline function _det(::Size, A::StaticMatrix,::Type)
+    return det(Matrix(A))
```

Comment:
Here you should be able to use a recent function I added - `drop_sdims(A)`, rather than allocating a `Matrix` with `Matrix(A)`.  Presumably this will be significantly faster.